cmds/app.c

```diff
@@ -123,28 +123,25 @@ static int cmd_appLoad(handler_t handler, size_t size, const char *name, char *i
 		return res;
 	}
 
-	switch (res) {
-		case dev_isNotMappable:
-			if ((entry = syspage_entryAdd(imaps, (addr_t)-1, size, SIZE_PAGE)) == NULL) {
-				log_error("\nCannot allocate memory for %s", name);
-				return -ENOMEM;
-			}
-
-			/* Copy elf file to selected entry */
-			if ((res = cmd_cp2ent(handler, entry)) < 0)
-				return res;
-			break;
-
-		case dev_isMappable:
-			if ((entry = syspage_entryAdd(NULL, addr + offs, size, SIZE_PAGE)) == NULL) {
-				log_error("\nCannot allocate memory for %s", name);
-				return -ENOMEM;
-			}
-			break;
-
-		default:
-			log_error("\nDevice mappable routine failed");
+	if (res == dev_isMappable || (res == dev_isNotMappable && (flags & flagSyspageNoCopy) != 0)) {
+		if ((entry = syspage_entryAdd(NULL, addr + offs, size, SIZE_PAGE)) == NULL) {
```

Comment:
You have changed some code to be compliant with some MISRA rules in the `devices`. I know that it is not a part of this PR but you can also update some code in this command as well. 

```suggestion
                entry = syspage_entryAdd(NULL, addr + offs, size, SIZE_PAGE)
		if (entry == NULL) {
```